web/src/components/tasks/NaicsCodeInput.tsx

```diff
@@ -101,15 +104,19 @@ export const NaicsCodeInput = (props: Props): ReactElement => {
   };
 
   const handleChange = (value: string): void => {
-    setNaicsCode(value);
-    if (value.length === LENGTH) {
-      if (getCode(value)) {
-        setIsInvalid(undefined);
+    if (isAuthenticated === IsAuthenticated.FALSE && business?.profileData.industryId === "generic") {
```

Comment:
```suggestion
    if (isAuthenticated === IsAuthenticated.FALSE {
```
[sand] We don't need to actually consider the industry type in this check. It can be limited to just checking for authentication.

Consider this, is the important part the industry selection or that the user is not registered? While this field will only display when the user's industry selection is "generic", we don't need to bake that into this check. We don't want to say "if you're _not_ logged in and your industry is _not_ generic then you _can_ enter content in this field".

Reply:
+1 This is a pebble, not sand, imo